File: cli/src/commands/me.ts
import chalk from 'chalk'
import { requireApiKey, getConfig, saveConfig } from '../config.js'
import { api } from '../api.js'
import { accent, accentBold, dim, setAccent } from '../theme.js'

interface MeData {
  pseudoHandle: string | null
  identityMode: string
  isVerified: boolean
  karmaScore: number
  activeSkinId: string
  customHex: string | null
  createdAt: string
}

export async function meCommand(): Promise<void> {
  const key = requireApiKey()
  const res = await api<MeData>('/me', key)
  const user = res.data!

  // Update local skin config
  const config = getConfig()
  config.skinId = user.activeSkinId
  if (user.customHex) config.customHex = user.customHex
  saveConfig(config)
  setAccent(user.activeSkinId, user.customHex)

  const name = user.pseudoHandle ?? dim('(no handle set)')
  const mode = user.identityMode
  const karma = user.karmaScore
  const joined = new Date(user.createdAt).toLocaleDateString('en-US', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  })

  console.log()
  console.log(accentBold(name))
  console.log()
  console.log(`  Identity:  ${accent(mode)}`)
  console.log(`  Verified:  ${chalk.green('Yes')}`)
  console.log(`  Karma:     ${accent(String(karma))}`)
  console.log(`  Skin:      ${accent(user.activeSkinId)}${user.customHex ? ` (${user.customHex})` : ''}`)
  console.log(`  Joined:    ${dim(joined)}`)
  console.log()
}

File: cli-rust/src/commands/me.rs
use anyhow::Result;
use serde::Deserialize;

use crate::api::ArkoraApi;
use crate::config;
use crate::theme;

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
struct MeData {
    pseudo_handle: Option<String>,
    identity_mode: String,
    karma_score: i64,
    active_skin_id: Option<String>,
    custom_hex: Option<String>,
    created_at: String,
}

pub async fn run(_color: (u8, u8, u8)) -> Result<()> {
    let cfg = config::load();
    let key = config::require_key(&cfg);
    let api = ArkoraApi::new(&config::api_url(&cfg), &key);

    let resp = api.get::<MeData>("/me").await?;
    let user = resp.data.unwrap();

    // Save skin to config for future sessions
    let mut cfg = config::load();
    cfg.skin_id = user.active_skin_id.clone();
    cfg.custom_hex = user.custom_hex.clone();
    config::save(&cfg)?;

    let skin = user.active_skin_id.as_deref().unwrap_or("monochrome");
    let color = theme::get_accent(skin, user.custom_hex.as_deref());

    let name = user
        .pseudo_handle
        .as_deref()
        .unwrap_or("(no handle set)");
    let joined = &user.created_at[..10]; // just the date

    println!();
    println!("{}", theme::accent_bold(name, color));
    println!();
    println!("  Identity:  {}", theme::accent(&user.identity_mode, color));
    println!("  Verified:  {}", theme::green("Yes"));
    println!(
        "  Karma:     {}",
        theme::accent(&user.karma_score.to_string(), color)
    );
    println!("  Skin:      {}", theme::accent(skin, color));
    println!("  Joined:    {}", theme::dim(joined));
    println!();

    Ok(())
}
